src/classify/shapetable.cpp

```diff
@@ -37,12 +37,14 @@ namespace tesseract {
 // Returns -1 if the unichar_id is not found
 int ShapeRating::FirstResultWithUnichar(const std::vector<ShapeRating> &results,
                                         const ShapeTable &shape_table, UNICHAR_ID unichar_id) {
-  for (unsigned r = 0; r < results.size(); ++r) {
-    const auto shape_id = results[r].shape_id;
+  size_t r = 0;
+  for (const auto &result : results) {
+    const auto shape_id = result.shape_id;
     const Shape &shape = shape_table.GetShape(shape_id);
     if (shape.ContainsUnichar(unichar_id)) {
```

Comment:
`if (shape_table.GetShape(result.shape_id).ContainsUnichar(unichar_id)) {`

also can be put inside `std::ranges::find_if()`